src/iterators.jl

```diff
@@ -168,6 +242,71 @@ Base.IteratorEltype(::Type{<:CellIterator}) = Base.HasEltype()
 Base.eltype(::Type{<:CellIterator{CC}}) where CC = CC
 Base.length(ci::CellIterator) = length(ci.set)
 
+##################
+## FaceIterator ##
+##################
+
+"""
+    FaceIterator(grid::Grid, interfaces_set=1:length(topology.face_skeleton), topology::ExclusiveTopology)
+    FaceIterator(dh::AbstractDofHandler, interfaces_set=1:length(topology.face_skeleton), topology::ExclusiveTopology)
+
+Create an `FaceIterator` to conveniently iterate over all, or a subset, of the faces in a
+face skeleton. The elements of the iterator are [`FaceCache`](@ref)s which are properly
+`reinit!`ialized. See [`FaceCache`](@ref) for more details.
+
+Looping over an `FaceIterator`, i.e.:
+```julia
+for fc in FaceIterator(grid, cellset, topology)
+    # ...
+end
+```
+is thus simply convenience for the following equivalent snippet:
+```julia
+fc = FaceCache(grid, topology)
+for face in topology.face_skeleton
+    neighbor_face = topology.face_neighbor[face[1], face[2]][1]
+    reinit!(fc, face, neighbor_face)
+    # ...
+end
+```
+!!! warning
+    `FaceIterator` is stateful and should not be used for things other than `for`-looping
+    (e.g. broadcasting over, or collecting the iterator may yield unexpected results).
+"""
+struct FaceIterator{Cache<:FaceCache, IC<:IntegerCollection}
+    cache::Cache
+    set::IC
+end
+
+function FaceIterator(gridordh::Union{Grid,AbstractDofHandler},
+                      set::Union{IntegerCollection,Nothing},
+                      topology::ExclusiveTopology)
+    if set === nothing
+        set = 1:length(topology.face_skeleton)
+    end
+    return FaceIterator(FaceCache(gridordh, topology), set)
+end
```

Comment:
If it's only for interfaces then we may want to exclude the boundary faces from within the iterator?
```julia
if set === nothing
  set = findall(face -> !isempty(topology.face_neighbor[face[1],face[2]]),topology.face_skeleton)
end
```